modify.py

```diff
@@ -47,23 +48,30 @@ def write_attribution_data(filepath, data):
 
         if cert_table_offset == 0 or cert_table_size == 0:
             # The file isn't signed
-            mapped.close()
-            return False
+            raise AttributionException('mapped is not signed')
 
         tag = b"__MOZCUSTOM__:"
         tag_index = mapped.find(tag, cert_table_offset,
             cert_table_offset + cert_table_size)
         if tag_index == -1:
-            mapped.close()
-            return False
+            raise AttributionException('mapped does not contain dummy cert')
 
         if sys.version_info >= (3,):
             data = data.encode("utf-8")
+
         mapped[tag_index+len(tag):tag_index+len(tag)+len(data)] = data
 
         return True
 
+    except AttributionException as e:
+        print(e)
+        return False
 
-if __name__ == "__main__":
-    write_attribution_data(sys.argv[1], sys.argv[2])
 
+if __name__ == "__main__":
+    # read mapped installer as bytearray so it can be modified
+    with open(sys.argv[1], 'r+b') as f:
+        mapped = bytearray(f.read())
+    write_attribution_data(mapped, sys.argv[2])
```

Comment:
sure, but that would have made the diff harder to read. i actually want to remove the try block entirely and just let it throw the exception.